db/write_batch.cc

```diff
@@ -1748,7 +1752,20 @@ class MemTableInserter : public WriteBatch::Handler {
           cfd->mem()->MarkFlushScheduled()) {
         // MarkFlushScheduled only returns true if we are the one that
         // should take action, so no need to dedup further
-        flush_scheduler_->ScheduleFlush(cfd);
+        flush_scheduler_->ScheduleWork(cfd);
+      }
+    }
+    // check if memtable_list size exceeds max_write_buffer_size_to_maintain
+    if (trim_history_scheduler_ != nullptr) {
+      auto* cfd = cf_mems_->current();
+      assert(cfd != nullptr);
+      if (cfd->ioptions()->max_write_buffer_size_to_maintain > 0 &&
+          cfd->mem()->ApproximateMemoryUsageCheap() +
+                  cfd->imm()->ApproximateMemoryUsageExcludingLast() >=
```

Comment:
`>=` or `>`?

Reply:
we compare against `max_write_buffer_size_to_maintain` in two places, here and `MemTableListVersion::MemtableLimitExceeded`. In both places we use `>=` which means once the size is reached, we will start trimming. I guess it's fine either way as long as we are consistent